### PR TITLE
chore: updating the CP version

### PR DESCRIPTION
**Pull request by bcm-at-zama**
Testing with the most recent CP
Maybe it will not be merged like that, but at least, should help to check there is no regression.

Also, @andrei-stoian-zama , if we want to check all the fixes are there, we can remove our workarounds in this PR and check. Don't hesitate to push in this branch, or we merge + team pushes in another PR 

Closes https://github.com/zama-ai/concrete-ml-internal/issues/3663
Closes https://github.com/zama-ai/concrete-ml-internal/issues/3978

### CODE REVIEW

**Review by RomanBredehoft — APPROVED** [review]

**Comment by fd0r** [comment]
Will Concrete Python deliver another version until the time we release? If not I would be in favor of not merging this.

**Comment by bcm-at-zama** [comment]
Anyway: we'll never deliver with nightlies, since they are not publicly available. And yes, we'll have official deliveries. But nightlies are ways for us to check everything is as expected, instead of waiting. So, if I were you, I'd check that the expected fixes are there, in this branch and PR

**Comment by bcm-at-zama** [comment]
FAILED tests/deployment/test_client_server.py::test_client_server_sklearn[3-KNeighborsClassifier] - RuntimeError: 

KNN failures as well

**Comment by bcm-at-zama** [comment]
@zama-ai/machine-learning apart the KNN problem, the last nightly CP looks good

**Review by fd0r — DISMISSED** [review]

**Comment by RomanBredehoft** [comment]
conflicting files

**Comment by bcm-at-zama** [comment]
yes but what do you want to do with this PR, first? are we going to merge it or not? if not, nothing to take care with the licenses. Are you going to test removing the workarounds in this PR, or elsewhere? @RomanBredehoft @andrei-stoian-zama

**Review by fd0r — DISMISSED** [review]

File: src/concrete/ml/sklearn/base.py [hunk 1]
```diff
@@ -1993,23 +1995,7 @@ def scatter1d(x, v, indices):
...
-            configuration = force_mono_parameter_in_configuration(args[1])
-            args_list = list(args)
-            args_list[1] = configuration
-            args = tuple(args_list)
-
-        # Else, retrieve the configuration in kwargs if it exists, or create a new one, and set the
-        # strategy to multi-parameter
-        else:
-            configuration = kwargs.get("configuration", None)
-            kwargs["configuration"] = force_mono_parameter_in_configuration(configuration)
```
Comment (RomanBredehoft) [inline src/concrete/ml/sklearn/base.py:2011]: I believe you can now remove `force_mono_parameter_in_configuration` from our source code !

**Comment by andrei-stoian-zama** [comment]
> yes but what do you want to do with this PR, first? are we going to merge it or not? if not, nothing to take care with the licenses. Are you going to test removing the workarounds in this PR, or elsewhere? @RomanBredehoft @andrei-stoian-zama

Yes, I also added the workaround removal , see the Closes section

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
this PR should remove `force_mono_parameter_in_configuration` from common/utils.py as it's not used anymore

File: src/concrete/ml/sklearn/base.py [hunk 2]
```diff
@@ -1993,23 +1994,7 @@ def scatter1d(x, v, indices):
 
         return numpy.expand_dims(topk_labels, axis=0)
 
-    # KNN works only for MONO in the latest concrete Python version
-    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3978
     def compile(self, *args, **kwargs) -> Circuit:
-        # If a configuration instance is given as a positional parameter, set the strategy to
-        # multi-parameter
-        if len(args) >= 2:
-            configuration = force_mono_parameter_in_configuration(args[1])
```
Comment (kcelia) [inline src/concrete/ml/sklearn/base.py:2002]: You need to delete the import of `from ..common.utils import force_mono_parameter_in_configuration` from base.py
And from src/concrete/ml/common/utils.py as well: 
````
from concrete.fhe import ParameterSelectionStrategy
from concrete.fhe.compilation.configuration import Configuration
```
+
```
def force_mono_parameter_in_configuration(configuration: Optional[Configuration], **kwargs) ....
````

**Comment by RomanBredehoft** [comment]
also I think you need to rebase @andrei-stoian-zama , you got a flaky that is currently skipped on main

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
@andrei-stoian-zama maybe you missed my last request for changes : 
> this PR should remove force_mono_parameter_in_configuration from common/utils.py as it's not used anymore

**Review by fd0r — DISMISSED** [review]

**Review by kcelia — APPROVED** [review]

**Review by fd0r — APPROVED** [review]

**Review by RomanBredehoft — APPROVED** [review]

**Pull request merged**